subiquity/server/server.py

```diff
@@ -551,7 +552,7 @@ async def middleware(self, request, handler):
                 resp.headers["x-error-msg"],
                 exc_info=exc,
             )
-            if not isinstance(exc, NonReportableException):
+            if not isinstance(exc, RecoverableError) or exc.produce_crash_report:
```

Comment:
The way this is written, if we ever throw a `NonReportableException` in a request handler then we'll get an apport report but we won't halt the installer, which is probably fine because we want to catch those and convert those to `RecoverableError` anyways?